prow/external-plugins/needs-rebase/plugin/plugin.go

```diff
@@ -379,3 +398,49 @@ func constructQueries(log *logrus.Entry, now time.Time, orgs, repos []string, us
 
 	return result
 }
+
+type timeNow func() time.Time
+
+type Cache struct {
+	cache       map[int]time.Time
+	validTime   time.Duration
+	currentTime timeNow
+	flush       func(cache *Cache)
+	mutex       sync.Mutex
+}
+
+func (cache *Cache) Get(key int) bool {
+	insertTime := cache.cache[key]
+	curTime := cache.currentTime()
+	age := curTime.Sub(insertTime)
+	if age > 0 && age < cache.validTime {
+		return true
+	}
+	cache.mutex.Lock()
+	delete(cache.cache, key)
+	cache.mutex.Unlock()
+	return false
+}
+
+func (cache *Cache) Set(key int) {
+	if cache.validTime > 0 {
+		cache.mutex.Lock()
+		cache.cache[key] = cache.currentTime()
+		cache.mutex.Unlock()
+	}
+}
+
+func (cache *Cache) Flush() {
+	cache.mutex.Lock()
+	cache.flush(cache)
+	cache.mutex.Unlock()
+}
+
+func NewCache(validTime int) *Cache {
+	return &Cache{
+		cache:       make(map[int]time.Time),
+		validTime:   time.Second * time.Duration(validTime),
+		currentTime: func() time.Time { return time.Now() },
+		flush:       func(cache *Cache) { cache.cache = make(map[int]time.Time) },
```

Comment:
Overriding the pointer should be locked with the mutex.

Reply:
It is locked in `func (cache *Cache) Flush()`

Reply:
Ok, but `flush` function itself is accessible from this context. Is it safe to lock it there not here? Another thing is that this lambda could be a function from the capital letter itself since the current `Flush()` is not doing much more. Although I would let someone more `Golang` experienced comment on that.